src/runtime/vdom/vdom-render.ts

```diff
@@ -985,25 +976,41 @@ render() {
         }
       }
 
-      for (i = 0; i < relocateNodes.length; i++) {
-        relocateData = relocateNodes[i];
-        nodeToRelocate = relocateData.$nodeToRelocate$;
+      for (const relocateData of relocateNodes) {
+        const nodeToRelocate = relocateData.$nodeToRelocate$;
         const slotRefNode = relocateData.$slotRefNode$;
 
         if (slotRefNode) {
-          // by default we're just going to insert it directly
-          // after the slot reference node
-          parentNodeRef = slotRefNode.parentNode;
-          insertBeforeNode = slotRefNode.nextSibling;
-          orgLocationNode = nodeToRelocate['s-ol'] as any;
-
-          while ((orgLocationNode = orgLocationNode.previousSibling as any)) {
-            refNode = orgLocationNode['s-nr'];
-            if (refNode && refNode['s-sn'] === nodeToRelocate['s-sn'] && parentNodeRef === refNode.parentNode) {
-              refNode = refNode.nextSibling as any;
-              if (!refNode || !refNode['s-nr']) {
-                insertBeforeNode = refNode;
-                break;
+          const parentNodeRef = slotRefNode.parentNode;
+          // When determining where to insert content, the most simple case would be
+          // to relocate the node immediately following the slot reference node. We do this
+          // by getting a reference to the node immediately following the slot reference node
+          // since we will use `insertBefore` to manipulate the DOM.
+          //
+          // If there is no node immediately following the slot reference node, then we will just
+          // end up appending the node as the last child of the parent.
+          let insertBeforeNode = slotRefNode.nextSibling as d.RenderNode;
```

Comment:
should this cast be to `d.RenderNode | null` ? I think if there's no sibling (which if I understand the comment correctly and the truthiness check below is possible) then I think `.nextSibling` would be `null`

Reply:
Good catch!